images/singleuser-sample/Dockerfile

```diff
@@ -1,11 +1,23 @@
-FROM jupyter/base-notebook:177037d09156
+FROM jupyter/base-notebook:8ccdfc1da8d5
 # Built from... https://hub.docker.com/r/jupyter/base-notebook/
 #               https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile
-# Built from... Ubuntu 16.04
+# Built from... Ubuntu 18.04
 
-# conda/pip/apt install additional packages here, if desired.
+# The jupyter/docker-stacks images contains jupyterhub, jupyterlab and the
+# jupyterlab-hub extension already.
+
+# Example install of git and nbgitpuller.
+# NOTE: git is already available in the jupyter/minimal-notebook image.
+USER root
+RUN apt-get update && apt-get install --yes \
+    git \
+ && rm -rf /var/lib/apt/lists/*
```

Comment:
hum, astonished there is not apt-get command for that...

Reply:
Yepp! Perhaps there is, I'm confused still about the things that is done by various `clean` commands. I've been afraid to use `apt-get` within a Dockerfile due to unknown cleanup procedures to make sure the image size does not explode as I thought it did when comparing. I had made a mistake of comparing actual size with Docker Hub's *compressed* size field though.

Since I needed to use `apt-get` now, I looked things up further and learned that this was the best practice on a ubuntu/debian based image. Apparently this is all the cleanup you need, and it is also important to have `apt-get update` run on the same line as `apt-get install` as if you don't, the `apt-get update` layer could be cached while `apt-get install` could be updated to get some new version that would not be found as `apt-get update` was not rerun.

Reply:
Thanks, I was roughly aware of all these issue, and still astonished there is no simpler way to fix those.